presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaConnectorModule.java

```diff
@@ -60,6 +60,7 @@ public void configure(Binder binder)
         binder.bind(KafkaFilterManager.class).in(Scopes.SINGLETON);
 
         configBinder(binder).bindConfig(KafkaConfig.class);
+        configBinder(binder).bindConfig(KafkaSecurityConfig.class);
```

Comment:
This should not be bind by default, but only when SSL is requested. It does make sense to allow users to configure truststore or keystore for SSL when they configure plain_text. Configuration has to be coherent otherwise it is hard to guess what was the actual desired behaviour.


See `io.prestosql.plugin.hive.security.HiveSecurityModule#bindSecurityModule` for example.

Reply:
Tried adapting `io.prestosql.plugin.hive.security.HiveSecurityModule#bindSecurityModule` but end up with the message `KafkaSecurityConfig is not explicitly bound` when it tries to inject both configs into the `constructor` of `Kafka*Factory`...

Pushed my current (broken) state - do you have a hint what I'm doing wrong?

Reply:
@kokosing Could you support on that?